File: examples/C++/rrclient.cpp
//   Request-reply client in C++
//   Connects REQ socket to tcp://localhost:5559
//   Sends "Hello" to server, expects "World" back
//

#include "zhelpers.hpp"
 
int main (int argc, char *argv[])
{
    zmq::context_t context(1);

	zmq::socket_t requester(context, ZMQ_REQ);
	requester.connect("tcp://localhost:5559");
 
	for( int request = 0 ; request < 10 ; request++) {
		
		s_send (requester, std::string("Hello"));
        std::string string = s_recv (requester);
		
		std::cout << "Received reply " << request 
				<< " [" << string << "]" << std::endl;
	}
}

File: examples/Perl/rrclient.pl
#!/usr/bin/perl
=pod

Hello World client

Connects REQ socket to tcp://localhost:5559

Sends "Hello" to server, expects "World" back

Author: Daisuke Maki (lestrrat)
Original version Author: Alexander D'Archangel (darksuji) <darksuji(at)gmail(dot)com>

=cut

use strict;
use warnings;
use 5.10.0;

use ZMQ::LibZMQ3;
use ZMQ::Constants qw(ZMQ_REQ);
use zhelpers;

my $context = zmq_init();

# Socket to talk to server
my $requester = zmq_socket($context, ZMQ_REQ);
zmq_connect($requester, 'tcp://localhost:5559');

for my $request_nbr (0 .. 9) {
    s_send($requester, 'Hello');
    my $string = s_recv($requester);
    say "Received reply $request_nbr [$string]";
}
